src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1684,7 +1684,7 @@ private static bool AccessingAutoPropertyFromConstructor(BoundExpression receive
             var propertyIsStatic = propertySymbol.IsStatic;
 
             return (object)sourceProperty != null &&
-                    sourceProperty.IsAutoPropertyWithGetAccessor &&
+                    sourceProperty.IsAutoProperty && sourceProperty.GetMethod is not null &&
```

Comment:
>sourceProperty.IsAutoProperty && sourceProperty.GetMethod is not null [](http://example.com/codeflow?start=20&length=69)

Same comment as for other places. Without taking a deep look, here is the theory that I have: "Here we actually want to check if an assignment to a property within constructor is supposed to assign to a backing field instead." If my theory is generally correct, let's add a new property with the specific name and purpose, and let's use it here. It would feel appropriate to rename the containing method accordingly as well. This is the process that we should go through for each API usage, one by one.